src/java/cern/accsoft/steering/jmad/model/JMadModelImpl.java

```diff
@@ -1,1308 +1,1308 @@
+            } catch (JMadModelException e) {
+                LOGGER.error("Error while passing new element-attribute value '" + name + "' to MadX.", e);
+            }
+        }
+    };
+
+    /**
+     * The listener which is added to all strengths in order to update the madx-instance, when one of the element values
+     * chages.
+     */
+    private final KnobListener strengthListener = new KnobListener() {
+        @Override
+        public void changedValue(Knob strength) {
+            try {
+                set(strength);
+            } catch (JMadModelException e) {
+                LOGGER.error("Error while passing new value for strength '" + strength.getName() + "' to MadX.", e);
+            }
+        }
+    };
+
+    /**
+     * the listener to be notified when the twiss-initial conditions are changed.
+     */
+    private final TwissListener twissListener = new TwissListener() {
+
+        /**
+         * @param twiss
+         */
+        @Override
+        public void changedTwiss(TwissInitialConditions twiss) {
+            dirtyModel = true;
+            fireBecameDirty();
+        }
+    };
+
+    public JMadModelImpl() {
+        twissInitialConditions.addListener(this.twissListener);
+        this.knobManager = new KnobManagerImpl(this);
+    }
+
+    /**
+     * ensures, that the model is initialized.
+     * 
+     * @throws JMadModelException if the initialization fails
+     */
+    private void ensureInit() throws JMadModelException {
+        if (!isInitialized()) {
+            LOGGER.debug("Model is not yet initialized. Starting init.");
+            init();
+        }
+    }
+
+    public void setModelDefinition(JMadModelDefinition modelDefinition) {
+        this.modelDefinition = modelDefinition;
+    }
+
+    @Override
+    public void init() throws JMadModelException {
+        LOGGER.debug("initializing model.");
+
+        /*
+         * first we have to start the kernel
+         */
+        try {
+            getKernel().start();
+        } catch (JMadException e) {
+            throw new JMadModelException("Error while initializing model.", e);
+        }
+
+        /*
+         * if we have no model definition then we are done
+         */
+        if (getModelDefinition() == null) {
+            return;
+        }
+
+        /*
+         * If we have a model definition then we first of all run all the init files.
+         */
+        processModelFiles(getModelDefinition().getInitFiles());
+
+        /* load the available beams for later use to the model */
+        for (SequenceDefinition sequenceDefinition : getModelDefinition().getSequenceDefinitions()) {
+            Beam beam = sequenceDefinition.getBeam();
+
+            if (beam != null) {
+                /* set sequence in beam for resbeam command */
+                beam.setSequence(sequenceDefinition.getName());
+                try {
+                    getKernel().execute(new SetBeam(beam));
+                } catch (JMa
```

Comment:
It is a bit strange to set this to true before the attributes are actually loaded ... but ok, I understand the need and anyway nothing is threadsafe ... horrible ;-) ... But ok ;-)